app/Controllers/StaticPagesController.js

```diff
@@ -0,0 +1,46 @@
+const catchAsync = require('../utils/catchAsync');
+const ApiError = require('../utils/ApiError.js');
+const httpStatus = require('http-status');
+
+const renderPageHandler = catchAsync(async (res, page) => {
+  const options = { root: process.cwd() };
+
+  // check if page should be rendered from a static file or should be redirected to another one
+  if (page.type === 'file') {
+    res.sendFile(page.location, options, (err) => {
+      if (err) {
+        throw new ApiError(httpStatus.NOT_FOUND, 'Page was not found');
+      }
+    });
+  } else {
```

Comment:
please check if type is redirect. This is much cleaner. If you want, you can add a default to the schema.